src/Utils.php

```diff
@@ -104,4 +104,19 @@ public static function decodeSocketId($socketId): ?object
         }
         return $socketIdObject;
     }
+
+    public static function isSameUser($token, $clientId)
+    {
+        // Decode the JWT token to extract the payload
+        $decodedToken = Utils::parseJwt($token);
+        $payload = $decodedToken['payload'];
+
+        // Check if the clientId in the payload matches the provided clientId
+        if (isset($payload['x-ably-clientId']) && $payload['x-ably-clientId'] == $clientId) {
```

Comment:
Make sure to have proper non-empty/non-null check for $payload['x-ably-clientId'], since guest user is allowed to upgrade to a specific user.